frontend-react/src/__mocks__/HistoryMockServer.ts

```diff
@@ -0,0 +1,35 @@
+import { rest } from "msw";
+import { setupServer } from "msw/node";
+
+import { HistoryApi } from "../network/api/History";
+import { Endpoint } from "../network/api/Api";
+
+/* TEST ENDPOINTS */
+const listEndpoint: Endpoint = HistoryApi.list();
+const detailSuccess: Endpoint = HistoryApi.detail("test");
+const detailFail: Endpoint = HistoryApi.detail("fail");
+
+const handlers = [
+    /* Successfully returns array of Reports */
+    rest.get(`http://localhost:3000${listEndpoint.url}`, (req, res, ctx) => {
+        return res(ctx.json(HistoryApi.testResponse(3)));
+    }),
+
+    /* Fails to return array of Reports with 404 */
+    rest.get(`http://localhost:3000${detailFail.url}/fail`, (req, res, ctx) => {
+        return res(ctx.status(404));
+    }),
+
+    /* Successfully returns detailed Report */
+    rest.get(`http://localhost:3000${detailSuccess.url}`, (req, res, ctx) => {
+        return res(ctx.json(HistoryApi.testResponse(1)));
+    }),
+
+    /* Fails to return detailed Report with 404 */
+    rest.get(`http://localhost:3000${detailFail.url}`, (req, res, ctx) => {
+        return res(ctx.status(404));
+    }),
```

Comment:
We should add a 5xx error

Reply:
Would that change what behavior we exhibit on the front-end? Right now, anything returned to a component has that custom response shape we defined, so I figured we could pipe the error and message over to our `ErrorPage` component and display it under the copy like `Status Code: (${status}) ${message}` or `Status Code: (500) Service Unavailable`. What this mock endpoint is doing is just ensuring we get a null data property back in that response shape:

```typescript
interface ResponseType<T> {
    loading: boolean;
    data?: T;
    status: number;
    message: string;
}
```